pkg/clients/rds/rds.go

```diff
@@ -135,70 +135,93 @@ func GenerateCreateDBInstanceInput(name, password string, p *v1alpha2.RDSInstanc
 }
 
 // GenerateModifyDBInstanceInput from RDSInstanceSpec
-func GenerateModifyDBInstanceInput(name string, p *v1alpha2.RDSInstanceParameters) *rds.ModifyDBInstanceInput {
+func GenerateModifyDBInstanceInput(name string, p *v1beta1.RDSInstanceParameters, currentState *rds.DBInstance) (*rds.ModifyDBInstanceInput, error) {
+	patch := &v1beta1.RDSInstanceParameters{}
+	if currentState != nil {
+		current := &v1beta1.RDSInstanceParameters{}
+		LateInitialize(current, *currentState)
+		currentJSON, err := json.Marshal(current)
+		if err != nil {
+			return nil, err
+		}
+		desiredJSON, err := json.Marshal(p)
+		if err != nil {
+			return nil, err
+		}
+		patchJSON, err := jsonpatch.CreateMergePatch(currentJSON, desiredJSON)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
https://github.com/imdario/mergo

You might also be able to use mergo to do this. It uses a whole bunch of reflection under the hood so it's likely not much better than JSON patching, but would at least avoid all the JSON serialisation.

Reply:
I've tried but it doesn't seem to have a mechanism for creating an intermediary _patch_ object to apply to other resources. Now, I looked up what controller-runtime uses for their new `Patch` call and it's the same as we do https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/patch.go#L61

I'll probably want to dig deeper into how to make this glue code more manageable in 0.5, either generation or reflection or json methods. Humans shouldn't be doing this work.